presto-main/src/main/java/com/facebook/presto/metadata/SessionPropertyManager.java

```diff
@@ -78,6 +84,16 @@ public SessionPropertyManager(List<PropertyMetadata<?>> systemSessionProperties)
         addSystemSessionProperties(systemSessionProperties);
     }
 
+    @Inject
+    public SessionPropertyManager(
+            SystemSessionProperties systemSessionProperties,
+            JavaFeaturesConfig javaFeaturesConfig,
+            NodeSpillConfig nodeSpillConfig)
+    {
+        this(systemSessionProperties.getSessionProperties());
+        this.systemSessionPropertyProviders.put("java-worker", new JavaWorkerSystemSessionPropertyProvider(javaFeaturesConfig, nodeSpillConfig));
```

Comment:
Let's create a Map in Guice that registers all built-in session property providers, and then just inject the map and put all of them.